tests/test_pgen.rs

```diff
@@ -175,3 +175,102 @@ fn test_grammar2() {
     has(&follow, "Q", vec!["+",")"]);
     has(&follow, "F", vec!["*", "+", ")"]);
 }
+
+fn lritem(lhs: &str, rhs: Vec<Symbol>, d: usize) -> LR1Item {
+    let mut item = LR1Item::new(lhs.to_string(), Vec::new(), d);
+    for e in rhs.iter() {
+        item.rhs.push(e.clone());
+    }
+    item
+}
+
+fn vec_to_set(vec: Vec<&str>) -> HashSet<String> {
+    let mut hs = HashSet::new();
+    for e in vec.iter() {
+        hs.insert(e.to_string());
+    }
+    hs
+}
+
+fn nt(s: &str) -> Symbol {
```

Comment:
I think we should use the macro, as the other test files do. I think you can just write !nonterminal("S") anyway, as `test_yaccparser.rs` does.